### PR TITLE
Dione Wilson Scrabble

### PR DESCRIPTION
**Pull request by dionew1**

### CODE REVIEW

File: lib/scrabble.rb [hunk 1]
```diff
@@ -15,4 +22,26 @@ def point_values
...
+    if word_length >= 7
+      10 + score
+    else
+      score
+    end
+  end
+
+
+  def score_with_mulitpliers(word, letter_multipliers, multiplier = 1)
+    #require 'pry'; binding.pry
```
Comment (VictoriaVasys) [inline lib/scrabble.rb:38]: Make sure to remove comments when pushing your final code.

File: lib/scrabble.rb [hunk 2]
```diff
@@ -1,7 +1,14 @@
 class Scrabble
 
   def score(word)
-    1
+    letter_values = point_values
+    if word == "" || word == nil
+      0
+    else
+      word.split("").map do |letter|
```
Comment (VictoriaVasys) [inline lib/scrabble.rb:8]: Like that you used map here! This method is a bit long though and could benefit from breaking this block out into its own method (or handle nil another way)

File: lib/scrabble.rb [hunk 3]
```diff
@@ -1,7 +1,14 @@
 class Scrabble
 
   def score(word)
-    1
+    letter_values = point_values
```
Comment (VictoriaVasys) [inline lib/scrabble.rb:4]: You don't need to assign this method return to a variable; since the method returns a hash, you can just access keys and values by calling the method where you need it

File: lib/scrabble.rb [hunk 4]
```diff
@@ -15,4 +22,26 @@ def point_values
       "Y"=>4, "Z"=>10
     }
   end
+
+  def greater_than_7_bonus(word)
```
Comment (VictoriaVasys) [inline lib/scrabble.rb:26]: Way to break out a separate method for the bonus!

File: lib/scrabble.rb [hunk 5]
```diff
@@ -15,4 +22,26 @@ def point_values
...
+
+
+  def score_with_mulitpliers(word, letter_multipliers, multiplier = 1)
+    #require 'pry'; binding.pry
+    letter_values = point_values
+    bonus = greater_than_7_bonus(word)
+    word_value = word.split("").each_with_index.map do |letter, index|
+      letter_values[letter.upcase] * letter_multipliers[index]
+    end
+    word_value.reduce(:+) * multiplier + (bonus * multiplier)
```
Comment (VictoriaVasys) [inline lib/scrabble.rb:44]: If you're using Ruby 2.4, use `#sum` instead of `#reduce`; it's a bit more readable

File: lib/scrabble.rb [hunk 6]
```diff
@@ -15,4 +22,26 @@ def point_values
...
+    word_length = word.length
+    if word_length >= 7
+      10 + score
+    else
+      score
+    end
+  end
+
+
+  def score_with_mulitpliers(word, letter_multipliers, multiplier = 1)
```
Comment (VictoriaVasys) [inline lib/scrabble.rb:37]: This method is a bit long too, push to break out logic.

File: test/scrabble_test.rb [hunk 7]
```diff
@@ -5,9 +5,54 @@
...
+    assert_equal 6, actual
+  end
+
+  def test_it_can_score_with_mutlipliers_and_multiplier
+    actual = Scrabble.new.score_with_mulitpliers("cat", [1,2,1], 2)
+
+    assert_equal 12, actual
+  end
+
+  def test_greater_than_7_bonus
```
Comment (VictoriaVasys) [inline test/scrabble_test.rb:45]: Love that you're testing your extra methods!

File: test/scrabble_test.rb [hunk 8]
```diff
@@ -5,9 +5,54 @@
...
+  end
+
+  def test_it_can_score_with_mutlipliers_and_7_letter_word
+    actual = Scrabble.new.score_with_mulitpliers('sparkle', [1,2,1,3,1,2,1], 2)
+
+    assert_equal 58, actual
+  end
+
+
 end
```
Comment (VictoriaVasys) [inline test/scrabble_test.rb:58]: Would love to see a few more tests to ensure functionality is what you expect (test different words, different multipliers, etc.)

File: test/scrabble_test.rb [hunk 9]
```diff
@@ -5,9 +5,54 @@
...
-    skip
     assert_equal 1, Scrabble.new.score("a")
     assert_equal 4, Scrabble.new.score("f")
   end
+
+  def test_it_can_score_empty_string
+    assert_equal 0, Scrabble.new.score("")
+  end
+
+  def test_it_can_score_nil
```
Comment (VictoriaVasys) [inline test/scrabble_test.rb:18]: I like your breakdown of testing- it's great to create separate tests for each edge case!

**Review by VictoriaVasys — COMMENTED** [review]
Great job overall Dione! I'd love to see you work on breaking out logic & adding tests to ensure your functionality is robust and dynamic. Also, make sure you're committing throughout (about every 15-20 minutes).
